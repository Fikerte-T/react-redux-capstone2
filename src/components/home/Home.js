import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getDataFromApi } from '../../redux/home/home';
import rightArrow from '../../images/arrow_circle_right_white_24dp.svg';

// const selectedData = (state) => state.homeReducer;
const Home = () => {
  const dispatch = useDispatch();
  const countryData = useSelector((state) => state.countries);
  const [searchValue, setSearchValue] = useState([]);

  const getData = () => {
    if (countryData.length === 0) {
      dispatch(getDataFromApi());
    }
  };
  useEffect(() => {
    getData();
  }, []);

  const date = new Date().toLocaleDateString('en-US', { month: 'short', day: 'numeric' });
  const handleSearch = (e) => {
    if (e.target.value === '') {
      setSearchValue([]);
    } else {
      const word = e.target.value;
      const filteredArr = countryData.filter((val) => val.country_name.toLowerCase()
        .includes(word.toLowerCase()));
      setSearchValue(filteredArr);
    }
  };
  return (
    <>
      <header>
        <div>
          <p className="date">{date}</p>
        </div>
        <div>
          <p data-testid="my-test-id">top confirmed cases</p>
        </div>
        <div>
          <input
            data-testid="input"
            type="text"
            placeholder="Search country"
            onChange={handleSearch}
          />
          {
            searchValue.length !== 0 && (
              <div className="search-wrapper">
                {searchValue.slice(0, 10).map((value) => (
                  <div className="country" key={value.country_id}>
                    <Link
                      to="/details"
                      state={value}
                    >
                      {value.country_name}
                    </Link>
                  </div>
                ))}
              </div>
            )
          }
        </div>
      </header>
      <div className="countries-wrapper">
        {
        countryData.sort((a, b) => (b.today_confirmed - a.today_confirmed))
          .slice(0, 20).map((c, index) => (
            <div key={c.country_id} className={`countries ${index % 2 === 0 ? 'light-color' : ''}`}>
              <Link to="/details" state={c}>
                <h2>{c.country_name.toUpperCase()}</h2>
                <p>{(c.today_confirmed).toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')}</p>
                <img src={rightArrow} alt="arrow to details" className="right-arrow" />
              </Link>
            </div>
          ))
      }
      </div>
    </>
  );
};

export default Home;
