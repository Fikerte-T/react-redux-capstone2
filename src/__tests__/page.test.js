import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import store from '../redux/configureStore';
import Home from '../components/home/Home';

describe('Components test', () => {
  it('Renders Home Page', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });

  it('has a text in the DOM', () => {
    const { container } = render(<Provider store={store}><Home /></Provider>);
    expect(container.getElementsByClassName('date').length).toBe(1);
  });
  it('has a text in the DOM', () => {
    render(<Provider store={store}><Home /></Provider>);
    expect(screen.getByTestId('my-test-id')).toHaveTextContent('top confirmed cases');
  });
  it('has a text in the DOM', () => {
    render(<Provider store={store}><Home /></Provider>);
    const input = screen.getByTestId('input');
    expect(input).toHaveAttribute('placeholder', 'Search country');
  });
});
