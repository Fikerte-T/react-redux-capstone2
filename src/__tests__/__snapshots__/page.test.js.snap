// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components test Renders Home Page 1`] = `
Array [
  <header>
    <div>
      <p
        className="date"
      >
        Feb 12
      </p>
    </div>
    <div>
      <p
        data-testid="my-test-id"
      >
        top confirmed cases
      </p>
    </div>
    <div>
      <input
        data-testid="input"
        onChange={[Function]}
        placeholder="Search country"
        type="text"
      />
    </div>
  </header>,
  <div
    className="countries-wrapper"
  />,
]
`;
